### DISCUSSION TITLE
Will you support onchain fallback inside Bolt 11 invoices?

### DISCUSSION THREAD

**Discussion by theDavidCoen** [btcpayserver/btcpayserver#3560] [Ideas / Feature Requests]
Do you guys plan to support onchain fallbacks INSIDE the bolt11 invoices' medatata?
Or will you just keep the current bip21 support with onchain fallback outside the bolt11 invoices?

https://lightning.money/spec/bolts/11.html

My question is just to know if the fallback inside the metadata is widely adopted and if you plan to support it too.
I believe the easiest solution is to put the fallback outside the bolt11, as you already do, in fact it seems it's considered a standard: https://bitcoinqr.dev/

Here I'm trying to do a general list to understand if it's indeed a standard: https://github.com/theDavidCoen/BIP21-URIs-with-Lightning-invoice-fallback-to-on-chain-support/blob/main/README.md

However, it is possible that some wallets/services will or already have a fallback in the metadata, in order to have a more elegant solution.
So I'm trying to understand what's the feeling about that and if it's in the roadmaps of different services, such as BTCpay server.

**Comment by pavlenex** [comment]
We've been supporting for over a year now. 

![Screenshot 2022-03-16 at 14 16 07](https://user-images.githubusercontent.com/36959754/158598224-3d6802c7-2d30-4fa5-a9c6-7197f0e010fe.png)


![Screenshot 2022-03-16 at 14 15 30](https://user-images.githubusercontent.com/36959754/158598108-22e84ee5-dc6a-4543-b799-de1ea6ee407b.png)

The bad news is that the UX is poor, since it's stilll shown weirdly in the checkout

![Screenshot 2022-03-16 at 14 15 34](https://user-images.githubusercontent.com/36959754/158598321-63c6fb36-5161-46ec-a405-3b6d3539f679.png)

**Comment by theDavidCoen** [comment]
Hey @pavlenex no that's an external fallback. I mean if you will add the fallback directly inside the Bolt11 tag.
Just to know if it's on your roadmap.

**Reply by dennisreimann** [reply]
@pavlenex The difference between the current solution and the one described above can be seen here: https://bitcoinqr.dev/#examples

**Reply by theDavidCoen** [reply]
Also, @pavlenex @dennisreimann here below an example of bolt11 with INTERNAL fallback:

`lnbcrt1p3ryggapp5l2puc46rh3mtl5x3g5jkgz46lpfavs2ue29jwuqkhw3uh6d4eepsdqqcqzpgfppq0x22pp8p4s9wkmc70vg5kr8xfhjfwq23sp50nh25xxwwyrf6n4tws42w5hrptsh72suzfzc5q93l3patny6e50s9qyyssqnk3asc39xnc9ntt8yxm5rqalupp4rka2qs0d6sg78xkw73gmt729h4kcefztk647qpll3c445j0dlr69a304g09dh9hkzqzsqt64l6spspk57l`

Decoded: https://lightningdecoder.com/lnbcrt1p3ryggapp5l2puc46rh3mtl5x3g5jkgz46lpfavs2ue29jwuqkhw3uh6d4eepsdqqcqzpgfppq0x22pp8p4s9wkmc70vg5kr8xfhjfwq23sp50nh25xxwwyrf6n4tws42w5hrptsh72suzfzc5q93l3patny6e50s9qyyssqnk3asc39xnc9ntt8yxm5rqalupp4rka2qs0d6sg78xkw73gmt729h4kcefztk647qpll3c445j0dlr69a304g09dh9hkzqzsqt64l6spspk57l

Json:
```
{
    "human_readable_part": {
        "prefix": "lnbcrt",
        "amount": "Any amount"
    },
    "data": {
        "time_stamp": 1647452445,
        "tags": [
            {
                "type": "p",
                "length": 52,
                "description": "payment_hash",
                "value": "fa83cc5743bc76bfd0d14525640abaf853d6415cca8b277016bba3cbe9b5ce43"
            },
            {
                "type": "d",
                "length": 0,
                "description": "description",
                "value": ""
            },
            {
                "type": "c",
                "length": 2,
                "description": "min_final_cltv_expiry",
                "value": 40
            },
            {
                "type": "f",
                "length": 33,
                "description": "fallback_address",
                "value": {
                    "version": 0,
                    "fallback_address": "0x22pp8p4s9wkmc70vg5kr8xfhjfwq23"
                }
            },
            {
                "type": "s",
                "length": 52,
                "description": "payment_secret",
                "value": "7ceeaa18ce71069d4eab742aa752e30ae17f2a1c12458a00b1fc43d5cc9acd1f"
            },
            {
                "type": "9",
                "length": 4,
                "description": "feature_bits",
                "value": "00100100001000000000"
            }
        ],
        "signature": {
            "r": "9da3d8622534f059ad6721b74183bfe04351dbaa041edd411e39acef451b5f94",
            "s": "5bd6d8ca44bb6abe007ff8e2b5a49edf8f45ec5f543cadb96f61005002f55fea",
            "recovery_flag": 1
        },
        "signing_data": "6c6e626372740c464423a10d3ea0f315d0ef1daff43451495902aebe14f5905732a2c9dc05aee8f2fa6d7390c1a0060020a121081e652821386b02badbc79ec452c339937925c054601a3e77550c6738834ea755ba1553a9718570bf950e0922c50058fe21eae64d668f8140424200"
    },
    "checksum": "spk57l"
}
```

Thanks https://github.com/siimone
